### PR TITLE
Introduce progress bar for visual feedback

### PR DESCRIPTION
**Pull request by atimmer**
This makes it easier when you are about to reach the limit of the title and meta description fields.

Before:
screen shot 2016-01-25 at 09 47 50

After:
screen shot 2016-01-25 at 09 48 48

@omarreiss and @jdevalk: Let me know what you think about the UX. I've tried to emulate the OSX progress bar.

### CODE REVIEW

**Comment by omarreiss** [comment]
Played with it a little bit, feels quite intuitive! However, it currently doesn't take templates into account. Please make sure it does.

**Comment by atimmer** [comment]
Ok, that requires https://github.com/Yoast/YoastSEO.js/pull/229. So I'll fix it when that's merged.

File: dist/yoast-seo.js [hunk 1]
```diff
@@ -3281,6 +3293,126 @@ function hasTrailingSlash( url ) {
...
+ */
+function rateTitleLength( titleLength ) {
+	var rating;
+
+	switch ( true ) {
+		case titleLength > 0 && titleLength <= 39:
+			rating = "ok";
+			break;
+
+		case titleLength >= 71:
```
Comment (andizer) [inline dist/yoast-seo.js]: Why not combining this with line 3334 like:
```
case titleLength > 0 && titleLength <= 39:
case titleLength >= 71:
     rating = "ok";
     break;		
```

File: dist/yoast-seo.js [hunk 2]
```diff
@@ -3281,6 +3293,126 @@ function hasTrailingSlash( url ) {
...
+ * Returns a rating based on the length of the meta description
+ *
+ * @param {string} metaDescLength
+ * @returns {string}
+ */
+function rateMetaDescLength( metaDescLength ) {
+	var rating;
+
+	switch ( true ) {
+		case metaDescLength > 0 && metaDescLength <= 120:
```
Comment (andizer) [inline dist/yoast-seo.js]: You can add the case (`case metaDescLength >= 157:`) on 3368 below this one.

**Comment by andizer** [comment]
CR done, just two comments.

**Comment by atimmer** [comment]
It now takes templates into account.

**Comment by terw-dan** [comment]
CR done :+1:

**Comment by jcomack** [comment]
Acceptance done

**Pull request merged**